iP

Wyrkx opened the pull request:
# Güliedistodiez

> "Due tomorrow do tomorrow." - Some guy

:calendar:

Güliedistodiez will help you keep track of various tasks. He is,
 - text-based
 - **easy** to use
 - *fast*

All you need to do is,
1. download the latest jar from [here](https://github.com/Wyrkx/ip/releases)
2. run it!

Features:
 - [X] Adding/removing tasks
 - [X] Searching for tasks by name
 - [ ] Searching for tasks by date

The `main` method or something
```
 public static void main(String[] args) {
     new Gulie(System.in, System.out).run();
 }
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

Wyrkx committed:
completed Level 0

Wyrkx committed:
completed Level 1

Wyrkx committed:
complated Level 2

Wyrkx committed:
completed level 3

Wyrkx committed:
completed level 4

Wyrkx committed:
completed level 4

Wyrkx committed:
completed automated text ui testing

Wyrkx committed:
completed level 5

Wyrkx committed:
completed level 6

Wyrkx committed:
Add save and load

Wyrkx committed:
Complete level 8

Wyrkx committed:
Complete A-MoreOOP

limcaaarl commented on src/main/java/Gulie.java:
```diff
@@ -0,0 +1,41 @@
+
+public class Gulie {
+    private GulieTasklist tasklist;
+    private final static String savepath = "./data/Gulie.txt";
```
Remember to follow the SCREAMING_SNAKE_CASE format for your constants 😁

```suggestion
    private final static String SAVE_PATH= "./data/Gulie.txt";
```

limcaaarl commented on src/main/java/GulieUi.java:
```diff
@@ -0,0 +1,81 @@
+
+public class GulieUi {
+    private static final String name = "Güliedistodiez";
+    private static final String line = "____________________________________________________________";
```
Remember to change these constants too!

```suggestion
    private static final String NAME = "Güliedistodiez";
    private static final String LINE = "____________________________________________________________";
```

limcaaarl commented on src/main/java/GulieStorage.java:
```diff
@@ -0,0 +1,52 @@
+            save.delete();
+            save.createNewFile();
+            FileWriter fw = new FileWriter(save);
+            for (Task t: tasklist) {
```
Remember to surround your operator with space characters here!
```suggestion
            for (Task t : tasklist) {
```

limcaaarl commented on src/main/java/Gulie.java:
```diff
@@ -0,0 +1,41 @@
+
+    public static void main(String[] args) {
+        new Gulie(System.in, System.out).run();
+    }
```
I like this part! It's very concise and clearly outlines the purpose of the main method.

limcaaarl reviewed (commented):
Good job! Overall, the code mostly adheres to the java coding standard with just a few minor tweaks needed. 👍👍👍

raysonchia commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,31 @@
+        this(name, by, false);
+    }
+
+    public Deadline(String name, LocalDateTime by, boolean mark) {
```
Perhaps a more boolean sounding variable name here? I noticed the same issue in several other places too.

raysonchia commented on src/main/java/GulieUi.java:
```diff
@@ -0,0 +1,81 @@
+import java.util.Scanner;
+
+public class GulieUi {
+    private static final String name = "Güliedistodiez";
```
I like how unique your bot name is!

raysonchia commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,61 @@
+                if (spl.length != 5) {
+                    throw new GulieException("This task is corrupted.");
+                }
+                return new Event(spl[2], LocalDateTime.parse(spl[3]), LocalDateTime.parse(spl[4]), spl[1].equals("1"));
```
Perhaps try to wrap your lines here? I noticed several other lines approaching the recommended soft limit too (110 chars).

raysonchia commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,61 @@
+    }
+
+    public static Task fromSaveString(String str) throws GulieException {
+        String[] spl = str.split("\t");
```
Perhaps a more intuitive variable name here?

raysonchia reviewed (commented):
Great job! I found most of your code well implemented. Just a few minor nits to enhance readability.

Wyrkx committed:
refactor into packages

Wyrkx committed:
Merge remote-tracking branch 'upstream/add-gradle-support'

Wyrkx committed:
Add gradle support

Wyrkx committed:
Add some junit tests

Wyrkx committed:
Build jar

Wyrkx committed:
Implement find

Wyrkx committed:
Tweak code to comply with standard

Wyrkx committed:
Add JavaDoc

Wyrkx committed:
Merge branch 'branch-A-CodingStandard'

Wyrkx committed:
Resolve conflict.

Wyrkx committed:
Updated README

Wyrkx committed:
Refactored to satisfy code guidelines

Wyrkx committed:
Refactored for better extensibility

Wyrkx committed:
Refactored for better extensibility

Wyrkx committed:
Add JavaFX GUI

Wyrkx committed:
Add assertions

Wyrkx committed:
Restructure code

Restructure startup sequence into different methods to enhance readability.

Wyrkx committed:
Restructure code

Add various tests to ensure code is working as expected. Fix bugs relating to improper restructuring done previously.

Wyrkx committed:
Add JavaDoc

The new classes in GUI are missing Javadoc and hence may confuse implementers on their functionality, hence add Javadoc to rectify.

Wyrkx committed:
Add assertions

Wyrkx committed:
Merge pull request #2 from Wyrkx/branch-A-CodeQuality

Add JavaDoc

Wyrkx committed:
Merge pull request #3 from Wyrkx/branch-A-Assertions

Add assertions

Wyrkx committed:
Implement schedule

Add a new command schedule that gets all tasks that coincide with the given date.

Wyrkx committed:
Add changes for better UX

Wyrkx committed:
Reenable GUI

Wyrkx committed:
Update save functionality
